tests/tlm/conftest.py

```diff
@@ -26,7 +32,59 @@ def tlm(studio: Studio) -> TLM:
     return studio.TLM()
 
 
+@pytest.fixture(scope="module")
+def tlm_dict(studio: Studio) -> Dict[str, Any]:
+    """Creates a dictionary of initialized tlm objects for each quality preset and model to be reused throughout the test.
+    Save randomly created options dictionary for each tlm object as well."""
+
+    tlm_dict = {}
+    for quality_preset in _VALID_TLM_QUALITY_PRESETS:
+        tlm_dict[quality_preset] = {}
+        for model in _VALID_TLM_MODELS:
+            tlm_dict[quality_preset][model] = {}
+            options = _get_options_dictionary(model)
+            tlm_dict[quality_preset][model]["tlm"] = studio.TLM(
+                quality_preset=quality_preset, options=options
+            )
+            tlm_dict[quality_preset][model]["options"] = options
+    return tlm_dict
+
+
 @pytest.fixture
 def tlm_rate_handler() -> TlmRateHandler:
     """Creates a TlmRateHandler with default settings."""
     return TlmRateHandler()
+
+
+def _get_options_dictionary(model: Optional[str]) -> dict:
+    if model is None:
+        options = {}
+    else:
+        options = {"model": model}
+
+    add_max_tokens = np.random.choice([True, False])
+    add_num_candidate_responses = np.random.choice([True, False])
+    add_num_consistency_samples = np.random.choice([True, False])
+    add_use_self_reflection = np.random.choice([True, False])
+    add_log_explanation = np.random.choice([True, False])
+    add_log_perplexity_score = np.random.choice([True, False])
+
+    if add_max_tokens:
+        options["max_tokens"] = int(np.random.randint(64, 512))
+    if add_use_self_reflection:
+        options["use_self_reflection"] = True
```

Comment:
I think you need the case where `use_self_reflection` is `False` as well

Reply:
This was giving huge typing errors (np._bool vs bool) so I removed it but good catch will add it back